### ISSUE TITLE
[QUERY] What is the .azstoragedml folder that gets created while blob upload? How to not create it?

### ISSUE THREAD

**Issue by himesh257** [Azure/azure-sdk-for-net#39002]
### Library name and version

Azure.Sotrage.DataMovement 12.0.0-beta.3, Azure.Sotrage.DataMovement.Blobs 12.0.0-beta.3

### Query/Question

While running below code, I see a folder .azstoragedml that gets created. I am trying to understand what this folder is and if it is possible to not create it.

```
BlockBlobClient destinationBlob = container.GetBlockBlobClient(relativeBlobReference);

TransferManagerOptions transferManagerOptions = new TransferManagerOptions
{
    MaximumConcurrency = GetThreadCount(blobStorageConfiguration, file, filesToUploadSizeMap)
};

TransferOptions singleTransferOptions = new TransferOptions();
singleTransferOptions.TransferStatus += (TransferStatusEventArgs args) =>
{
    if (args.StorageTransferStatus == StorageTransferStatus.Completed)
    {
        log($"Transfer completed for TransferId: {args.TransferId}");
    }

    return Task.CompletedTask;
};
singleTransferOptions.TransferSkipped += (TransferSkippedEventArgs args) =>
{
    log($"Transfer skipped for TransferId: {args.TransferId}," +
        $"Source Resource: {args.SourceResource.Path}, +" +
        $"Destination Resource: {args.DestinationResource.Path}");

    return Task.CompletedTask;
};
singleTransferOptions.TransferFailed += (TransferFailedEventArgs args) =>
{
    log($"Exception occured with TransferId: {args.TransferId}," +
        $"Source Resource: {args.SourceResource.Path}, +" +
        $"Destination Resource: {args.DestinationResource.Path}," +
        $"Exception Message: {args.Exception.Message}");

    var azureBlobStorageException = new AzureBlobStorageException(args.Exception.Message, args.Exception);
    uploadExceptions.Add(azureBlobStorageException);

    return Task.CompletedTask;
};

TransferManager transferManager = new TransferManager(transferManagerOptions);

DataTransfer dataTransfer = await transferManager.StartTransferAsync(
    sourceResource: new LocalFileStorageResource(file),
    destinationResource: new BlockBlobStorageResource(destinationBlob),
    transferOptions: singleTransferOptions).ConfigureAwait(continueOnCapturedContext: false);

 await dataTransfer.AwaitCompletion();
```

### Environment

windows 10

**Comment by amnguye** [comment]
Sorry for the late reply.

This folder is created in order to support the feature for Pause and Resume (e.g. pausing a transfer and resuming it). In order to be able to resume a transfer, we have to store the state of the transfer, which is in the ".azstoragedml" folder.

It's not an unreasonable ask to disable this feature, I will add an option to disable the feature so this will not appear for the user.

**Comment by jalauzon-msft** [comment]
Hi @himesh257 Himesh, we just merged #46286 which adds the option to disable the checkpointing (pause/resume) feature entirely so this folder will not be created. This will be available in our next release and will be done with `TransferCheckpointStoreOptions.Disabled()`.

Just as a note, if you wish to have the pause/resume functionality there is also an option to configure the location that this folder is created. You can look at [CheckpointerOptions](https://learn.microsoft.com/en-us/dotnet/api/azure.storage.datamovement.transfermanageroptions.checkpointeroptions?view=azure-dotnet-preview#azure-storage-datamovement-transfermanageroptions-checkpointeroptions) inside [TransferManagerOptions ](https://learn.microsoft.com/en-us/dotnet/api/azure.storage.datamovement.transfermanageroptions?view=azure-dotnet-preview)for more details. Do note, this class will have some breaking changes in the upcoming release. Thanks.
